File: app/src/main/java/io/dwak/holohackernews/app/ui/storydetail/StoryDetailActivity.java
package io.dwak.holohackernews.app.ui.storydetail;

import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import butterknife.ButterKnife;
import butterknife.InjectView;
import io.dwak.holohackernews.app.R;
import io.dwak.holohackernews.app.preferences.UserPreferenceManager;
import io.dwak.holohackernews.app.ui.storylist.MainActivity;
import me.imid.swipebacklayout.lib.SwipeBackLayout;
import tv.acfun.a63.swipe.SwipeAppCompatActivity;

public class StoryDetailActivity extends SwipeAppCompatActivity {

    @InjectView(R.id.toolbar) Toolbar mToolbar;
    private StoryDetailFragment mStoryDetailFragment;
    private boolean mSaved;
    private boolean mFromExternal;

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_story_detail);
        ButterKnife.inject(this);
        Intent intent = getIntent();
        long storyId = 0;
        mFromExternal = false;
        if (intent != null) {
            Bundle extras = intent.getExtras();
            if (extras != null) {
                storyId = extras.getLong(MainActivity.STORY_ID);
                mSaved = extras.getBoolean(StoryDetailFragment.LOADING_FROM_SAVED);
            }

            final Uri data = intent.getData();
            if (data != null) {
                storyId = Long.parseLong(data.getQueryParameter("id"));
                mFromExternal = true;
            }
        }
        if (mToolbar != null) {
            mToolbar.setTitleTextColor(getResources().getColor(android.R.color.white));
            setSupportActionBar(mToolbar);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        if(UserPreferenceManager.getInstance().isSwipeBackEnabled()) {
            getSwipeBackLayout().setEdgeTrackingEnabled(SwipeBackLayout.EDGE_LEFT);
        }
        else{
            getSwipeBackLayout().setEnableGesture(false);
        }
        if (savedInstanceState == null) {
            mStoryDetailFragment = StoryDetailFragment.newInstance(storyId, mSaved);
            getSupportFragmentManager().beginTransaction()
                                       .add(R.id.container, mStoryDetailFragment)
                                       .commit();
        }
    }

    @Override
    public void onBackPressed() {
        if (mStoryDetailFragment != null && mStoryDetailFragment.isLinkViewVisible() && !UserPreferenceManager.getInstance().showLinkFirst()) {
            mStoryDetailFragment.hideLinkView();
        }
        else {
            super.onBackPressed();
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                finishAfterTransition();
            }
            else {
                overridePendingTransition(R.anim.fadein, R.anim.view_right_to_offscreen);
            }
        }
    }
}

File: app/src/main/kotlin/io/dwak/holohackernews/app/ui/detail/view/StoryDetailActivity.kt
package io.dwak.holohackernews.app.ui.detail.view

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import com.r0adkll.slidr.Slidr
import com.r0adkll.slidr.model.SlidrConfig
import com.r0adkll.slidr.model.SlidrPosition
import io.dwak.holohackernews.app.R
import io.dwak.holohackernews.app.extension.getLong
import io.dwak.holohackernews.app.extension.navigateTo
import io.dwak.holohackernews.app.extension.with

class StoryDetailActivity : AppCompatActivity(){

    companion object {
        val ITEM_ID = "ITEM_ID"

        fun newIntent(context : Context, itemId : Long) : Intent {
            return Intent(context, StoryDetailActivity::class.java).with {
                putExtra(ITEM_ID, itemId)
            }
        }
    }

    override fun onCreate(savedInstanceState : Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_story_detail)
        setSupportActionBar(findViewById(R.id.toolbar) as Toolbar)
        supportActionBar.setDisplayHomeAsUpEnabled(true)
        Slidr.attach(this, SlidrConfig.Builder().edge(true)
                .position(SlidrPosition.LEFT)
                .build())
        navigateTo(StoryDetailFragment.newInstance(getLong(ITEM_ID)), addToBackStack = false)
    }

    override fun onOptionsItemSelected(item : MenuItem?) : Boolean {
        when(item?.itemId){
            android.R.id.home -> onBackPressed()
            else -> {}
        }
        return super.onOptionsItemSelected(item)
    }
}